operator/controllers/kyma_controller.go

```diff
@@ -121,283 +122,282 @@ func (r *KymaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
+		return ctrl.Result{RequeueAfter: r.RequeueIntervals.Failure}, r.HandleProcessingState(ctx, &kyma)
 	case operatorv1alpha1.KymaStateDeleting:
-		if dependentsDeleting, err := r.HandleDeletingState(ctx, &logger, &kyma); err != nil {
+		if dependentsDeleting, err := r.HandleDeletingState(ctx, &kyma); err != nil {
 			return ctrl.Result{}, err
 		} else if dependentsDeleting {
 			return ctrl.Result{RequeueAfter: r.RequeueIntervals.Waiting}, nil
 		}
 	case operatorv1alpha1.KymaStateError:
-		return ctrl.Result{RequeueAfter: r.RequeueIntervals.Waiting}, r.HandleErrorState(ctx, &logger, &kyma)
+		return ctrl.Result{RequeueAfter: r.RequeueIntervals.Waiting}, r.HandleErrorState(ctx, &kyma)
 	case operatorv1alpha1.KymaStateReady:
-		return ctrl.Result{RequeueAfter: r.RequeueIntervals.Success}, r.HandleReadyState(ctx, &logger, &kyma)
+		//TODO Adjust again
+		return ctrl.Result{RequeueAfter: r.RequeueIntervals.Success}, r.HandleReadyState(ctx, &kyma)
 	}
 
 	return ctrl.Result{}, nil
 }
 
-func (r *KymaReconciler) HandleInitialState(ctx context.Context, _ *logr.Logger, kyma *operatorv1alpha1.Kyma) error {
-	return r.updateKymaStatus(ctx, kyma, operatorv1alpha1.KymaStateProcessing, "initial state")
+func (r *KymaReconciler) HandleInitialState(ctx context.Context, kyma *operatorv1alpha1.Kyma) error {
+	return status.Helper(r).UpdateStatus(ctx, kyma, operatorv1alpha1.KymaStateProcessing, "initial state")
 }
 
-func (r *KymaReconciler) HandleProcessingState(ctx context.Context, logger *logr.Logger, kyma *operatorv1alpha1.Kyma) error {
+func (r *KymaReconciler) HandleProcessingState(ctx context.Context, kyma *operatorv1alpha1.Kyma) error {
+	logger := log.FromContext(ctx)
 	logger.Info("processing " + kyma.Name)
 
+	if len(kyma.Spec.Components) < 1 {
+		return fmt.Errorf("no component specified for resource %s", kyma.Name)
+	}
+
 	// fetch templates
 	templates, err := release.GetTemplates(ctx, r, kyma)
 	if err != nil {
-		return r.updateKymaStatus(ctx, kyma, operatorv1alpha1.KymaStateError, "templates could not be fetched")
+		return status.Helper(r).UpdateStatus(ctx, kyma, operatorv1alpha1.KymaStateError, "templates could not be fetched")
 	}
 
-	// reconcile from templates
-	if err := r.ReconcileKymaForRelease(ctx, kyma, templates); err != nil {
+	// these are the actual modules
+	modules, err := util.ParseTemplates(kyma, templates)
+	if err != nil {
 		return err
 	}
 
-	// update status conditions
-	updateRequired, err := r.checkAndUpdateComponentConditions(ctx, kyma, templates)
+	statusUpdateRequiredFromCreation, err := r.CreateOrUpdateModules(ctx, kyma, modules)
 	if err != nil {
+		message := fmt.Sprintf("Component CR creation error: %s", err.Error())
+		logger.Info(message)
+		r.Event(kyma, "Warning", "ReconciliationFailed", fmt.Sprintf("Reconciliation failed: %s", message))
+		statusErr := status.Helper(r).UpdateStatus(ctx, kyma, operatorv1alpha1.KymaStateError, message)
+		if statusErr != nil {
+			return statusErr
+		}
```

Comment:
this way the external `err` is separated from the one below. But your wish :) 
```suggestion
		if err := status.Helper(r).UpdateStatus(ctx, kyma, operatorv1alpha1.KymaStateError, message); err != nil {
			return err
		}
```